src/test/java/redis/clients/jedis/tests/JedisSentinelPoolTest.java

```diff
@@ -161,6 +164,42 @@ public void customClientName() {
 
     assertTrue(pool.isClosed());
   }
+  
+  @Test
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(null, 0, 2000, 2000, "foobared", 0, "my_shiny_client_name");
+    
+    try(JedisSentinelPool pool = new JedisSentinelPool(MASTER_NAME, sentinels, new JedisPoolConfig(), factory);
+        Jedis obj1 = pool.getResource(); ) {
+      obj1.set("foo", "bar");
+      assertEquals("bar", obj1.get("foo"));
+      assertEquals(1, pool.getNumActive());
+      
+      factory.setPassword("wrong password");
+      try (Jedis obj2 = pool.getResource();) {
+        fail("Should not get resource from pool");
+      }catch (JedisConnectionException e) {}
+    }
+  }
+ 
+  @Test
+  public void testResetValidPassword() {
+    JedisFactory factory = new JedisFactory(null, 0, 2000, 2000, "wrong password", 0, "my_shiny_client_name");
+    JedisSentinelPool pool = new JedisSentinelPool(MASTER_NAME, sentinels, new JedisPoolConfig(), factory);
+    Jedis obj = null;
+    try {
+      pool.getResource();
```

Comment:
Get this resource and close it properly, just to avoid any possible connection leak. try-with-resources can be used for convenience.